### ISSUE TITLE
electron-builder install-app-deps fails.

### ISSUE THREAD

**Issue by jean-ben-guigui** [jitsi/jitsi-meet-electron#112]
Hello,

I am quite excited about this new electron app.
Nevertheless, I am not able to make it work on a windows 7 computer. I managed to make it work on my friend windows 10 computer though. I'd really appreciate your help.

I am failing at the first step of the process : npm install.
I get the following error (with set DEBUG=electron-builder):

> electron@3.0.0-beta.5 postinstall E:\git\jitsi-meet-electron\node_modules\electron
> node install.js

> jitsi-meet-electron@1.0.0-alpha.3 postinstall E:\git\jitsi-meet-electron
> electron-builder install-app-deps

  • electron-builder version=20.28.1
  • loaded configuration file=package.json ("build" field)
  • unresolved dependencies after first round unresolved=fsevents
  • rebuilding native production dependencies platform=win32 arch=ia32
  • spawning        command=C:\Program Files (x86)\nodejs\node.exe C:\Program Files (x86)\nodejs\node_modules\npm\bin\npm-cli.js rebuild jitsi-meet-electron-utils@0.5.2 robotjs@0.4.7 cwd=E:\git\jitsi-meet-electron

> jitsi-meet-electron-utils@0.5.2 install E:\git\jitsi-meet-electron\node_modules\jitsi-meet-electron-utils
> prebuild-install || node-gyp rebuild

E:\git\jitsi-meet-electron\node_modules\node-abi\index.js:30
  throw new Error('Could not detect abi for version ' + target + ' and runtime ' + runtime + '.  Updating "node-abi" might help solve this issue if it is a new release of ' + runtime)
  ^

Error: Could not detect abi for version 3.0.0-beta.5 and runtime electron.  Updating "node-abi" might help solve this issue if it is a new release of electron
    at getAbi (E:\git\jitsi-meet-electron\node_modules\node-abi\index.js:30:9)
    at module.exports (E:\git\jitsi-meet-electron\node_modules\prebuild-install\rc.js:64:12)
    at Object.<anonymous> (E:\git\jitsi-meet-electron\node_modules\prebuild-install\bin.js:8:25)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)

E:\git\jitsi-meet-electron\node_modules\jitsi-meet-electron-utils>if not defined npm_config_node_gyp (node "C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild )
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
  index.cc
  sourceId2Coordinates.cc
  win_delay_load_hook.cc
C:\Users\aduver\.electron-gyp\iojs-3.0.0-beta.5\deps\v8\include\v8.h(2509): error C2144: syntax error : 'int' should be preceded by ';' (..\node_addons\sourceId2Coordinates\src\index.cc) [E:\git\jitsi-me
et-electron\node_modules\jitsi-meet-electron-utils\build\sourceId2Coordinates.vcxproj]
C:\Users\aduver\.electron-gyp\iojs-3.0.0-beta.5\deps\v8\include\v8.h(2509): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int (..\node_addons\sourceId2Coordinates\
src\index.cc) [E:\git\jitsi-meet-electron\node_modules\jitsi-meet-electron-utils\build\sourceId2Coordinates.vcxproj]
C:\Users\aduver\.electron-gyp\iojs-3.0.0-beta.5\deps\v8\include\v8.h(4124): error C2610: 'v8::WasmCompiledModule::TransferrableModule::TransferrableModule(v8::WasmCompiledModule::TransferrableModule &&)'
 : is not a special member function which can be defaulted (..\node_addons\sourceId2Coordinates\src\index.cc) [E:\git\jitsi-meet-electron\node_modules\jitsi-meet-electron-utils\build\sourceId2Coordinates
.vcxproj]
C:\Users\aduver\.electron-gyp\iojs-3.0.0-beta.5\deps\v8\include\v8.h(4127): error C2610: 'v8::WasmCompiledModule::TransferrableModule &v8::WasmCompiledModule::TransferrableModule::operator =(v8::WasmComp
iledModule::TransferrableModule &&)' : is not a special member function which can be defaulted (..\node_addons\sourceId2Coordinates\src\index.cc) [E:\git\jitsi-meet-electron\node_modules\jitsi-meet-elect
ron-utils\build\sourceId2Coordinates.vcxproj]
C:\Users\aduver\.electron-gyp\iojs-3.0.0-beta.5\deps\v8\include\v8.h(4215): error C2610: 'v8::WasmModuleObjectBuilderStreaming::WasmModuleObjectBuilderStreaming(v8::WasmModuleObjectBuilderStreaming &&)'
: is not a special member function which can be defaulted (..\node_addons\sourceId2Coordinates\src\index.cc) [E:\git\jitsi-meet-electron\node_modules\jitsi-meet-electron-utils\build\sourceId2Coordinates.
vcxproj]
C:\Users\aduver\.electron-gyp\iojs-3.0.0-beta.5\deps\v8\include\v8.h(4220): error C2610: 'v8::WasmModuleObjectBuilderStreaming &v8::WasmModuleObjectBuilderStreaming::operator =(v8::WasmModuleObjectBuilde
rStreaming &&)' : is not a special member function which can be defaulted (..\node_addons\sourceId2Coordinates\src\index.cc) [E:\git\jitsi-meet-electron\node_modules\jitsi-meet-electron-utils\build\sourc
eId2Coordinates.vcxproj]
C:\Users\aduver\.electron-gyp\iojs-3.0.0-beta.5\deps\v8\include\v8.h(4510): error C2146: syntax error : missing ';' before identifier 'size_t' (..\node_addons\sourceId2Coordinates\src\index.cc) [E:\git\j
itsi-meet-electron\node_modules\jitsi-meet-electron-utils\build\sourceId2Coordinates.vcxproj]
C:\Users\aduver\.electron-gyp\iojs-3.0.0-beta.5\deps\v8\include\v8.h(4510): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int (..\node_addons\sourceId2Coordinates\
src\index.cc) [E:\git\jitsi-meet-electron\node_modules\jitsi-meet-electron-utils\build\sourceId2Coordinates.vcxproj]
C:\Users\aduver\.electron-gyp\iojs-3.0.0-beta.5\src\node_buffer.h(34): error C2597: illegal reference to non-static member 'v8::TypedArray::kMaxLength' (..\node_addons\sourceId2Coordinates\src\index.cc)
[E:\git\jitsi-meet-electron\node_modules\jitsi-meet-electron-utils\build\sourceId2Coordinates.vcxproj]
C:\Users\aduver\.electron-gyp\iojs-3.0.0-beta.5\src\node_buffer.h(34): error C2440: 'initializing' : cannot convert from 'unknown' to 'const unsigned int' (..\node_addons\sourceId2Coordinates\src\index.c
c) [E:\git\jitsi-meet-electron\node_modules\jitsi-meet-electron-utils\build\sourceId2Coordinates.vcxproj]
          There is no context in which this conversion is possible
gyp ERR! build error
gyp ERR! stack Error: `msbuild` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:262:23)
gyp ERR! stack     at emitTwo (events.js:126:13)
gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd E:\git\jitsi-meet-electron\node_modules\jitsi-meet-electron-utils
gyp ERR! node -v v8.11.4
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! jitsi-meet-electron-utils@0.5.2 install: `prebuild-install || node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the jitsi-meet-electron-utils@0.5.2 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\aduver\AppData\Roaming\npm-cache\_logs\2018-08-27T16_17_39_207Z-debug.log
  • exited          command=node.exe code=1 pid=18684
Error: C:\Program Files (x86)\nodejs\node.exe exited with code 1
    at ChildProcess.childProcess.once.code (E:\git\jitsi-meet-electron\node_modules\builder-util\src\util.ts:254:14)
    at Object.onceWrapper (events.js:317:30)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at maybeClose (internal/child_process.js:925:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
From previous event:
    at rebuild (E:\git\jitsi-meet-electron\node_modules\app-builder-lib\out\util\yarn.js:239:18)
    at E:\git\jitsi-meet-electron\node_modules\app-builder-lib\src\util\yarn.ts:20:11
From previous event:
    at installOrRebuild (E:\git\jitsi-meet-electron\node_modules\app-builder-lib\out\util\yarn.js:68:17)
    at E:\git\jitsi-meet-electron\node_modules\electron-builder\src\cli\install-app-deps.ts:56:9
    at Generator.next (<anonymous>)
From previous event:
    at installAppDeps (E:\git\jitsi-meet-electron\node_modules\electron-builder\out\cli\install-app-deps.js:174:17)
    at then (E:\git\jitsi-meet-electron\node_modules\electron-builder\src\cli\cli.ts:42:48)
    at runCallback (timers.js:810:20)
    at tryOnImmediate (timers.js:768:5)
    at processImmediate [as _immediateCallback] (timers.js:745:5)
From previous event:
    at Object.args [as handler] (E:\git\jitsi-meet-electron\node_modules\electron-builder\src\cli\cli.ts:42:48)
    at Object.runCommand (E:\git\jitsi-meet-electron\node_modules\yargs\lib\command.js:237:44)
    at Object.parseArgs [as _parseArgs] (E:\git\jitsi-meet-electron\node_modules\yargs\yargs.js:1059:30)
    at Object.get [as argv] (E:\git\jitsi-meet-electron\node_modules\yargs\yargs.js:1000:21)
    at Object.<anonymous> (E:\git\jitsi-meet-electron\node_modules\electron-builder\src\cli\cli.ts:25:28)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)
    at bootstrap_node.js:612:3
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! jitsi-meet-electron@1.0.0-alpha.3 postinstall: `electron-builder install-app-deps`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the jitsi-meet-electron@1.0.0-alpha.3 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\aduver\AppData\Roaming\npm-cache\_logs\2018-08-27T16_17_39_390Z-debug.log

**Comment by akshitkrnagpal** [comment]
Hey @jean-ben-guigui 
Can you verify the version of `node-abi` installed?

Try `npm ls node-abi`

From a quick look at the [changelog](https://github.com/lgeiger/node-abi/releases) electron 3.0 was added in v2.4.2.

**Comment by jean-ben-guigui** [comment]
I had a v2.4.1 node-abi. So I upgraded to 2.4.3, now it seams that I have a problem with electron gyp, I get the following error message:

 • rebuilding native production dependencies platform=win32 arch=ia32
  • spawning        command=C:\Program Files (x86)\nodejs\node.exe C:\Program Files (x86)\nodejs\node_modules\npm\bin\npm-cli.js rebuild jitsi-meet-electron-utils@0.5.2 robotjs@0.4.7 cwd=E:\git\jitsi-meet-electron

> jitsi-meet-electron-utils@0.5.2 install E:\git\jitsi-meet-electron\node_modules\jitsi-meet-electron-utils
> prebuild-install || node-gyp rebuild

prebuild-install WARN install No prebuilt binaries found (target=3.0.0-beta.5 runtime=electron arch=ia32 platform=win32)

E:\git\jitsi-meet-electron\node_modules\jitsi-meet-electron-utils>if not defined npm_config_node_gyp (node "C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild )
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
  index.cc
  sourceId2Coordinates.cc
  win_delay_load_hook.cc
C:\Users\aduver\.electron-gyp\iojs-3.0.0-beta.5\deps\v8\include\v8.h(2509): error C2144: syntax error : 'int' should be preceded by ';' (..\node_addons\sourceId2Coordinates\src\index.cc) [E:\git\jitsi-me
et-electron\node_modules\jitsi-meet-electron-utils\build\sourceId2Coordinates.vcxproj]
C:\Users\aduver\.electron-gyp\iojs-3.0.0-beta.5\deps\v8\include\v8.h(2509): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int (..\node_addons\sourceId2Coordinates\
src\index.cc) [E:\git\jitsi-meet-electron\node_modules\jitsi-meet-electron-utils\build\sourceId2Coordinates.vcxproj]
C:\Users\aduver\.electron-gyp\iojs-3.0.0-beta.5\deps\v8\include\v8.h(4124): error C2610: 'v8::WasmCompiledModule::TransferrableModule::TransferrableModule(v8::WasmCompiledModule::TransferrableModule &&)'
 : is not a special member function which can be defaulted (..\node_addons\sourceId2Coordinates\src\index.cc) [E:\git\jitsi-meet-electron\node_modules\jitsi-meet-electron-utils\build\sourceId2Coordinates
.vcxproj]
C:\Users\aduver\.electron-gyp\iojs-3.0.0-beta.5\deps\v8\include\v8.h(4127): error C2610: 'v8::WasmCompiledModule::TransferrableModule &v8::WasmCompiledModule::TransferrableModule::operator =(v8::WasmComp
iledModule::TransferrableModule &&)' : is not a special member function which can be defaulted (..\node_addons\sourceId2Coordinates\src\index.cc) [E:\git\jitsi-meet-electron\node_modules\jitsi-meet-elect
ron-utils\build\sourceId2Coordinates.vcxproj]
C:\Users\aduver\.electron-gyp\iojs-3.0.0-beta.5\deps\v8\include\v8.h(4215): error C2610: 'v8::WasmModuleObjectBuilderStreaming::WasmModuleObjectBuilderStreaming(v8::WasmModuleObjectBuilderStreaming &&)'
: is not a special member function which can be defaulted (..\node_addons\sourceId2Coordinates\src\index.cc) [E:\git\jitsi-meet-electron\node_modules\jitsi-meet-electron-utils\build\sourceId2Coordinates.
vcxproj]
C:\Users\aduver\.electron-gyp\iojs-3.0.0-beta.5\deps\v8\include\v8.h(4220): error C2610: 'v8::WasmModuleObjectBuilderStreaming &v8::WasmModuleObjectBuilderStreaming::operator =(v8::WasmModuleObjectBuilde
rStreaming &&)' : is not a special member function which can be defaulted (..\node_addons\sourceId2Coordinates\src\index.cc) [E:\git\jitsi-meet-electron\node_modules\jitsi-meet-electron-utils\build\sourc
eId2Coordinates.vcxproj]
C:\Users\aduver\.electron-gyp\iojs-3.0.0-beta.5\deps\v8\include\v8.h(4510): error C2146: syntax error : missing ';' before identifier 'size_t' (..\node_addons\sourceId2Coordinates\src\index.cc) [E:\git\j
itsi-meet-electron\node_modules\jitsi-meet-electron-utils\build\sourceId2Coordinates.vcxproj]
C:\Users\aduver\.electron-gyp\iojs-3.0.0-beta.5\deps\v8\include\v8.h(4510): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int (..\node_addons\sourceId2Coordinates\
src\index.cc) [E:\git\jitsi-meet-electron\node_modules\jitsi-meet-electron-utils\build\sourceId2Coordinates.vcxproj]
C:\Users\aduver\.electron-gyp\iojs-3.0.0-beta.5\src\node_buffer.h(34): error C2597: illegal reference to non-static member 'v8::TypedArray::kMaxLength' (..\node_addons\sourceId2Coordinates\src\index.cc)
[E:\git\jitsi-meet-electron\node_modules\jitsi-meet-electron-utils\build\sourceId2Coordinates.vcxproj]
C:\Users\aduver\.electron-gyp\iojs-3.0.0-beta.5\src\node_buffer.h(34): error C2440: 'initializing' : cannot convert from 'unknown' to 'const unsigned int' (..\node_addons\sourceId2Coordinates\src\index.c
c) [E:\git\jitsi-meet-electron\node_modules\jitsi-meet-electron-utils\build\sourceId2Coordinates.vcxproj]
          There is no context in which this conversion is possible
gyp ERR! build error
gyp ERR! stack Error: `msbuild` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:262:23)
gyp ERR! stack     at emitTwo (events.js:126:13)
gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd E:\git\jitsi-meet-electron\node_modules\jitsi-meet-electron-utils
gyp ERR! node -v v8.11.4
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! jitsi-meet-electron-utils@0.5.2 install: `prebuild-install || node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the jitsi-meet-electron-utils@0.5.2 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\aduver\AppData\Roaming\npm-cache\_logs\2018-08-28T07_13_46_976Z-debug.log
  • exited          command=node.exe code=1 pid=15444
Error: C:\Program Files (x86)\nodejs\node.exe exited with code 1
    at ChildProcess.childProcess.once.code (E:\git\jitsi-meet-electron\node_modules\builder-util\src\util.ts:254:14)
    at Object.onceWrapper (events.js:317:30)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at maybeClose (internal/child_process.js:925:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
From previous event:
    at rebuild (E:\git\jitsi-meet-electron\node_modules\app-builder-lib\out\util\yarn.js:239:18)
    at E:\git\jitsi-meet-electron\node_modules\app-builder-lib\src\util\yarn.ts:20:11
From previous event:
    at installOrRebuild (E:\git\jitsi-meet-electron\node_modules\app-builder-lib\out\util\yarn.js:68:17)
    at E:\git\jitsi-meet-electron\node_modules\electron-builder\src\cli\install-app-deps.ts:56:9
    at Generator.next (<anonymous>)
From previous event:
    at installAppDeps (E:\git\jitsi-meet-electron\node_modules\electron-builder\out\cli\install-app-deps.js:174:17)
    at then (E:\git\jitsi-meet-electron\node_modules\electron-builder\src\cli\cli.ts:42:48)
    at runCallback (timers.js:810:20)
    at tryOnImmediate (timers.js:768:5)
    at processImmediate [as _immediateCallback] (timers.js:745:5)
From previous event:
    at Object.args [as handler] (E:\git\jitsi-meet-electron\node_modules\electron-builder\src\cli\cli.ts:42:48)
    at Object.runCommand (E:\git\jitsi-meet-electron\node_modules\yargs\lib\command.js:237:44)
    at Object.parseArgs [as _parseArgs] (E:\git\jitsi-meet-electron\node_modules\yargs\yargs.js:1059:30)
    at Object.get [as argv] (E:\git\jitsi-meet-electron\node_modules\yargs\yargs.js:1000:21)
    at Object.<anonymous> (E:\git\jitsi-meet-electron\node_modules\electron-builder\src\cli\cli.ts:25:28)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)
    at bootstrap_node.js:612:3
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! jitsi-meet-electron@1.0.0-alpha.3 postinstall: `electron-builder install-app-deps`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the jitsi-meet-electron@1.0.0-alpha.3 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\aduver\AppData\Roaming\npm-cache\_logs\2018-08-28T07_13_47_085Z-debug.log

**Comment by akshitkrnagpal** [comment]
Hey @jean-ben-guigui 

I tried to reproduce this error in Windows 7 32-bit. But it actually worked.

Did you install `windows-build-tools`?

`npm install -g --production windows-build-tools`

If this doesn't work try to download and install Visual C++ build tools. (Give it a google search)

**Comment by saghul** [comment]
Tentatively closing. Thanks for helping out @akshitkrnagpal !

**Comment by jean-ben-guigui** [comment]
I followed these steps:
- installed nodeJs
- `npm install -g node-gyp`
- `npm install --global --production windows-build-tools`
- `git clone https://github.com/jitsi/jitsi-meet-electron.git`
- npm i

I installed C++ build tools from VS2017. But I got the same error logs. Running out of option on that computer :(
Works as a charm on MacOS though. I'll try to make it work when I'll have some times.I'll be posting back here if success there is.

Thanks for your help and that great work on electron.
